### ISSUE TITLE
AWS::Lambda::Function inlineCode support for Ruby

### ISSUE THREAD

**Issue by kstys** [aws-cloudformation/cloudformation-coverage-roadmap#951]
### Name of the resource

AWS::Lambda::Function

### Description

Manually creating a Lambda function in AWS allows you to pick from NodeJS, Python, Ruby, and some others. As it stands for CloudFormation templates, AWS::Lambda::Function can not have the property `Runtime: ruby2.7` (or older versions) and use inline code via the property:
```YAML
Code: 
   ZipFile: |
      # code
```

This issue https://github.com/aws-cloudformation/cloudformation-coverage-roadmap/issues/80 has garnered enough attention that I believe it is worthy of implementing inline Ruby code if possible.

**Comment by bananastalktome** [comment]
@cfn-github-issues-bot got my hopes up. Where in the roadmap is this @prerna-p ? Looks like it was "We're working on it" 2021-11-18 and "Coming Soon" 2022-03-04, but is it all the way back to "Researching" now?

**Comment by prerna-p** [comment]
I will defer to @micahduron from AWS Lambda team to provide insight on the timelines for this. Thank you reaching out!

**Comment by jtuliani** [comment]
Defining Lambda function code inline within a CloudFormation template is currently only supported for Node and Python runtimes, as described in our documentation at https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-lambda-function-code.html.

We do not currently have any plans to extend inline CloudFormation functions to support Ruby. The issue #80 cited above has only one request for Ruby support; the bulk of the discussion relates to Python and Node.

**Comment by kstys** [comment]
@jtuliani Is lack of number of requests the only reason Ruby support will not be supported for CloudFormation templates? I counted 9 unique interactions with this post and ruby interactions in #80.

The main reason I created this request is because AWS Lambda in the UI supports creating Lambda functions in Ruby while CloudFormation templates do not. This creates an inconsistency between the resource creation available in the UI and the resource creation and maintenance via automatic deployments.
